Segfault encountered during merge in Snowflake

running into this error in latest version `stable-v0.19.1`
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0x362034f]

goroutine 2871425 [running]:
github.com/PeerDB-io/peer-flow/connectors/snowflake.(*mergeStmtGenerator).generateMergeStmt(0xc0264362d0, {0xc045d8331c, 0xe})
        /root/flow/connectors/snowflake/merge_stmt_generator.go:31 +0xaf
github.com/PeerDB-io/peer-flow/connectors/snowflake.(*SnowflakeConnector).mergeTablesForBatch.func1()
        /root/flow/connectors/snowflake/snowflake.go:560 +0x71
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /go/pkg/mod/golang.org/x/sync@v0.9.0/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 2867236
        /go/pkg/mod/golang.org/x/sync@v0.9.0/errgroup/errgroup.go:75 +0x96
```

--- a/flow/connectors/snowflake/snowflake.go
+++ b/flow/connectors/snowflake/snowflake.go
@@ -245,6 +245,7 @@ func (c *SnowflakeConnector) getDistinctTableNamesInBatch(
 	ctx context.Context,
 	flowJobName string,
 	batchId int64,
+	tableToSchema map[string]*protos.TableSchema,
 ) ([]string, error) {
 	rawTableIdentifier := getRawTableIdentifier(flowJobName)
 
@@ -261,7 +262,9 @@ func (c *SnowflakeConnector) getDistinctTableNamesInBatch(
 		if err := rows.Scan(&result); err != nil {
 			return nil, fmt.Errorf("failed to read row: %w", err)
 		}
-		destinationTableNames = append(destinationTableNames, result.String)
+		if _, ok := tableToSchema[result.String]; ok {
+			destinationTableNames = append(destinationTableNames, result.String)
+		}
 	}
 
 	if err := rows.Err(); err != nil {
@@ -520,7 +523,7 @@ func (c *SnowflakeConnector) mergeTablesForBatch(
 	tableToSchema map[string]*protos.TableSchema,
 	peerdbCols *protos.PeerDBColumns,
 ) error {
-	destinationTableNames, err := c.getDistinctTableNamesInBatch(ctx, flowName, batchId)
+	destinationTableNames, err := c.getDistinctTableNamesInBatch(ctx, flowName, batchId, tableToSchema)
 	if err != nil {
 		return err
 	}

--- a/flow/connectors/snowflake/snowflake.go
+++ b/flow/connectors/snowflake/snowflake.go
@@ -264,6 +264,8 @@ func (c *SnowflakeConnector) getDistinctTableNamesInBatch(
 		}
 		if _, ok := tableToSchema[result.String]; ok {
 			destinationTableNames = append(destinationTableNames, result.String)
+		} else {
+			c.logger.Warn("table not found in table to schema mapping", "table", result.String)
 		}
 	}
 

